routes/posts.js

```diff
@@ -0,0 +1,111 @@
+var Bcrypt = require('bcrypt');
+var Joi = require('joi');
+var Auth = require('./auth');
+
+exports.register = function(server, options, next){
+
+  server.route([
+    // POST a post
+    {
+      method: 'POST',
+      path: '/posts',
+      config: {
+        handler: function(request, reply){
+          Auth.authenticated(request, function(result){
+            if (result.authenticated) {
+              var db       = request.server.plugins['hapi-mongodb'].db;
+              var session  = request.session.get('hapi_twitter_session');
+              var ObjectId = request.server.plugins['hapi-mongodb'].ObjectID;
+
+              
+              db.collection('users').findOne({ _id: result.user_id }, function(err, user) {
+                if (err) {return reply('Internal MongoDB error', err);}
+
+                var post = {
+                  'title':    request.payload.post.title,
+                  'content':  request.payload.post.content,
+                  'category': request.payload.post.category,
+                  'photo':    request.payload.post.photo,
+                  'username': user.username,
+                  'user_id':  ObjectId(result.user_id)
+                };
+
+                db.collection('posts').insert(post, function(err,writeResult){
+                  if (err) {return reply('Internal MongoDB error', err);}
+                  
+                  reply(writeResult);
+                });
+              });
+            } else {
+              reply(result.message);
+            }
+          })
+        },
+        validate: {
+          payload: {
+            post: {
+              content:    Joi.string().min(1).max(140).required(),
+              title:      Joi.string().min(1).max(140).required(),
+              category:   Joi.string().min(1).max(140).required(),
+              photo:      Joi.string().min(1).max(140).allow()
+            }
+          }
+        }
+      }
+    },
+    // GET all posts
+    {
+      method: 'GET',
+      path: "/posts",
+      handler: function(request, reply){
+        var db = request.server.plugins['hapi-mongodb'].db;
+        db.collection('posts').find().toArray(function(err,posts){
+          if (err) {return reply('Internal MongoDB error', err);}
+          reply(posts);
+        })
+      }
+    },
+    // GET one post
+    {
+      method: 'GET',
+      path: '/posts/{id}',
+      handler: function(request, reply){
+        var post_id  = encodeURIComponent(request.params.id);
+        // .id retrieve from path , path from window's browser url? 
```

Comment:
you can't retrieve from window browser url, because this code executes in the backend